core-bundle/tests/EventListener/PreviewToolbarListenerTest.php

```diff
@@ -0,0 +1,345 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Tests\EventListener;
+
+use Contao\CoreBundle\EventListener\PreviewToolbarListener;
+use Contao\CoreBundle\Routing\ScopeMatcher;
+use Contao\CoreBundle\Tests\TestCase;
+use PHPUnit\Framework\MockObject\MockObject;
+use Symfony\Component\HttpFoundation\HeaderBag;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Routing\RequestContext;
+use Symfony\Component\Routing\RouterInterface;
+use Twig\Environment;
+
+class PreviewToolbarListenerTest extends TestCase
+{
+    /**
+     * @dataProvider getInjectToolbarData
+     */
+    public function testInjectToolbar($content, $expected): void
```

Comment:
This seems to be the same test as `testToolbarIsInjected()`?

Reply:
One test is checking the bare snipped, i.e. if the snippet is correct, while the other test is checking whether the toolbar is injected on valid response.